tasks/componentUpdateTasks.ts

```diff
@@ -0,0 +1,186 @@
+
+    // Extract version from file name
+    // Example: "Microsoft.VisualStudio.Copilot.Roslyn.LanguageServer-18.0.671-alpha.zip"
+    let version: string | null = null;
+    const m = zipName.match(/Microsoft\.VisualStudio\.Copilot\.Roslyn\.LanguageServer-(.+)\.zip$/i);
+    if (m && m[1]) {
+        version = m[1];
+    }
+
+    if (!version) {
+        console.log(`Could not extract version from file name ${zipName}; skipping.`);
+        return;
+    }
+
+    console.log(`Extracted version: ${version}`);
+
+    const pkg = getPackageJSON();
+    let updated = false;
+
+    if (pkg.runtimeDependencies && Array.isArray(pkg.runtimeDependencies)) {
+        for (let i = 0; i < pkg.runtimeDependencies.length; i++) {
+            const dep = pkg.runtimeDependencies[i];
+            if (dep && dep.id === 'RoslynCopilot') {
+                const oldUrl = dep.url as string;
+                const newUrl = oldUrl.replace(
+                    /Microsoft\.VisualStudio\.Copilot\.Roslyn\.LanguageServer-[^/]+?\.zip/,
+                    `Microsoft.VisualStudio.Copilot.Roslyn.LanguageServer-${version}.zip`
+                );
+                if (oldUrl !== newUrl) {
+                    pkg.runtimeDependencies[i].url = newUrl;
+                    updated = true;
+                    console.log(`Updated RoslynCopilot url:\n  ${oldUrl}\n-> ${newUrl}`);
+                } else {
+                    console.log('RoslynCopilot url already up to date.');
+                }
+                break;
+            }
+        }
+    }
+
+    if (!updated) {
+        console.log('No changes required to package.json; aborting PR creation.');
+        return;
+    }
+
+    // Persist package.json
+    fs.writeFileSync('package.json', JSON.stringify(pkg, null, 2) + '\n', { encoding: 'utf8' });
+
+    // Prepare git
+    const safeVersion = version.replace(/[^A-Za-z0-9_.-]/g, '-');
+    const branch = `update/roslyn-copilot-${safeVersion}`;
+
+    // Make this optional so it can be tested locally by using dev's information. In real CI user name and email are always supplied.
+    if (parsedArgs.userName) {
+        await git(['config', '--local', 'user.name', parsedArgs.userName]);
+    }
+    if (parsedArgs.email) {
+        await git(['config', '--local', 'user.email', parsedArgs.email]);
+    }
+
+    await git(['checkout', '-b', branch]);
+    await git(['add', 'package.json']);
+    await git(['commit', '-m', `chore: update RoslynCopilot url to ${version}`]);
+
+    const pat = process.env['GitHubPAT'];
+    if (!pat) {
+        throw 'No GitHub PAT found.';
+    }
+
+    const remoteRepoAlias = 'targetRepo';
+    await git(
+        ['remote', 'add', remoteRepoAlias, `https://${parsedArgs.userName}:${pat}@github.com/dotnet/vscode-csharp.git`],
+        // Note: don't print PAT to console
+        false
+    );
+    await git(['fetch', remoteRepoAlias]);
+
+    const lsRemote = await git(['ls-remote', remoteRepoAlias, 'refs/head/' + branch]);
```

Comment:
Extract to function or add comment that this is checking if a particular branch exists on the remote.